inc/mqttlog.class.php

```diff
@@ -99,4 +104,83 @@ protected function saveMqttMessage($direction, $topicList, $msg) {
          $this->addToDB();
       }
    }
+
+   function getTabNameForItem(CommonGLPI $item, $withtemplate = 0) {
+      if (!self::canView()) {
+         return '';
+      }
+
+      switch ($item->getType()) {
+         case PluginFlyvemdmAgent::class:
+            if (!$withtemplate) {
+               $nb = 0;
+               $topic = $item->getTopic();
+               if ($_SESSION['glpishow_count_on_tabs'] && $topic) {
+                  $DbUtil = new DbUtils();
+                  $nb = $DbUtil->countElementsInTable(
+                     static::getTable(),
+                     ['topic' => ['LIKE', $topic . '%']]
+                  );
+               }
+               return self::createTabEntry(self::getTypeName(Session::getPluralNumber()), $nb);
+            }
+            break;
+         case PluginFlyvemdmFleet::class:
+            if (!$withtemplate) {
+               $nb = 0;
+               $topic = $item->getTopic();
+               if ($_SESSION['glpishow_count_on_tabs'] && $topic) {
+                  $DbUtil = new DbUtils();
+                  $nb = $DbUtil->countElementsInTable(
+                     static::getTable(),
+                     ['topic' => ['LIKE', $topic . '%']]
+                  );
+               }
+               return self::createTabEntry(self::getTypeName(Session::getPluralNumber()), $nb);
+            }
+            break;
+      }
+   }
+
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum = 1, $withtemplate = 0) {
+      switch (get_class($item)) {
+         case PluginFlyvemdmAgent::class:
+         case PluginFlyvemdmFleet::class:
+            self::showMqttLogs($item);
+            return true;
+            break;
+      }
+   }
+
+   public static function showMqttLogs(PluginFlyvemdmNotifiableInterface $item) {
+      if (!self::canView()) {
+         return '';
+      }
+
+      $start = isset($_GET["start"]) ? intval($_GET["start"]) : 0;
+
+      // get items
+      $rows = [];
+      $topic = $item->getTopic();
+      if($topic){
+         $mqttLogs = new self;
+         $rows = $mqttLogs->find("topic LIKE '" . $topic . "%'", 'DATE DESC');
+      }
+      $number = count($rows);
+
+      // get the pager
+      $pager = Html::printAjaxPager(self::getTypeName(1), $start, $number, '', false);
+
+      $data = [
+         'empty_msg' => sprintf(__('You have not recorded any %1$s yet'), self::getTypeName(1)),
```

Comment:
So you don't like the text "You have not recorded any MQTT subscriber yet" as is used as general message for GLPI?

Reply:
Yes, because an admin does not adds anything in mqtt logs. This would lead him to think he heeds to add entries to see something here. Actually he cannot do that directly. He can do that by deploying policies. Also other messages are initiated by devices.

I think we can find something more accurate.

Reply:
Check it again,  I set the default message from GLPI